### PR TITLE
Consolidate ca2007 tests

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

**Comment by Youssef1313** [comment]
@sharwell I haven't completed this yet, but currently facing an exception from the testing library.

> System.ArgumentException : The input contained more than one start tag for span '#1' (Parameter 'input')

Could you give it a look please?

File: src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotDirectlyAwaitATaskTests.cs [hunk 1]
```diff
@@ -1,62 +1,20 @@
...
 namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.UnitTests
 {
     public class DoNotDirectlyAwaitATaskTests
     {
-        [Theory]
-        [InlineData("Task")]
-        [InlineData("Task<int>")]
-        [InlineData("ValueTask")]
-        [InlineData("ValueTask<int>")]
-        public async Task CSharpSimpleAwaitTask(string typeName)
```
Comment (sharwell) [inline src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotDirectlyAwaitATaskTests.cs:23]: 💡 Can you keep the tests in this file in the same relative order they were in? One of the general rules to follow in writing analyzer tests is each test which reports a diagnostic should include the fixed code and be written as a code fix test. Keeping them in the same order makes it easier to see the application of this rule and simplifies the review a bit.

Comment (Youssef1313) [reply src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotDirectlyAwaitATaskTests.cs:23]: @sharwell Do you mean that tests which produce no diagnostic should come after ones that produces a diagnostic and a codefix?

**Comment by Evangelink** [comment]
@Youssef1313 if the PR is still making sense, could you fix the conflict?

**Comment by sharwell** [comment]
@Youssef1313 Sorry I totally missed this PR when I submitted https://github.com/dotnet/roslyn-analyzers/pull/4894 ☹️

**Comment by Youssef1313** [comment]
> @Youssef1313 Sorry I totally missed this PR when I submitted #4894 ☹️

Even I missed it when I saw your PR :smile:

**Pull request closed without merging**